save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TestRepository.kt

```diff
@@ -14,8 +14,10 @@ interface TestRepository : BaseEntityRepository<Test> {
      *
      * @param limit
      * @param offset
+     * @param executionId
      * @return List of Tests
      */
-    @Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' limit ?1 offset ?2", nativeQuery = true)
-    fun retrieveBatches(limit: Int, offset: Int): List<Test>
+    @Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' " +
```

Comment:
What about this giant query? I think we already have entities with proper relations, so all this joins can be generated by hibernate. And there are [several ways](https://www.baeldung.com/jpa-limit-query-results) to control result set size

Reply:
Deleted this method and added method to TestExecutionRepository.